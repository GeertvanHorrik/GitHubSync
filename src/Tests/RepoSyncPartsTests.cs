using GitHubSync;
using ObjectApproval;
using Xunit;
using Xunit.Abstractions;

public class RepoSyncPartsTests : TestBase
{
    [Fact]
    public void Simple()
    {
        var repoSync = BuildRepoSync();
        repoSync.AddBlob("added1");
        repoSync.AddBlob("added2","target2");
        repoSync.RemoveBlob("removed1");
        repoSync.RemoveBlob("removed2", "target2");
        repoSync.AddTarget("owner1", "repo1", "branch1");
        repoSync.AddTarget("owner2", "repo2", "branch2");
        Verify(repoSync);
    }
    [Fact]
    public void AddBlob()
    {
        var repoSync = BuildRepoSync();
        repoSync.AddBlob("added1");
        repoSync.AddBlob("added2","target2");
        repoSync.AddBlob("sourceDir/added3","targetDir/target3");
        Verify(repoSync);
    }
    [Fact]
    public void AddTree()
    {
        var repoSync = BuildRepoSync();
        repoSync.AddSourceItem(TreeEntryTargetType.Tree, "added1");
        repoSync.AddSourceItem(TreeEntryTargetType.Tree,"added2","target2");
        repoSync.AddSourceItem(TreeEntryTargetType.Tree,"sourceDir/added3","targetDir/target3");
        Verify(repoSync);
    }

    [Fact]
    public void RemoveBlob()
    {
        var repoSync = BuildRepoSync();
        repoSync.RemoveBlob("added1");
        repoSync.RemoveBlob("added2","target2");
        repoSync.RemoveBlob("sourceDir/added3","targetDir/target3");
        Verify(repoSync);
    }

    [Fact]
    public void AddTarget()
    {
        var repoSync = BuildRepoSync();
        repoSync.AddTarget("repo1");
        repoSync.AddTarget("owner2","repo2","branch2");
        Verify(repoSync);
    }

    static RepoSync BuildRepoSync()
    {
        var credentials = CredentialsHelper.Credentials;
        return new RepoSync(credentials, "owner", "GitHubSync.TestRepository", "source");
    }

    static void Verify(RepoSync repoSync)
    {
        ObjectApprover.VerifyWithJson(
            new
            {
                repoSync.itemsToSync,
                repoSync.targets
            });
    }

    public RepoSyncPartsTests(ITestOutputHelper output) : base(output)
    {
    }
}