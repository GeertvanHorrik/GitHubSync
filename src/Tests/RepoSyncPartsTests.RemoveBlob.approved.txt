{
  itemsToSync: [
    {
      Parts: {
        Owner: 'owner',
        Repository: 'GitHubSync.TestRepository',
        Branch: 'source',
        Path: 'added1',
        Name: 'added1',
        NumberOfPathSegments: 1,
        Url: 'https://github.com/owner/GitHubSync.TestRepository/blob/source/added1',
        ParentTreePart: {
          Owner: 'owner',
          Repository: 'GitHubSync.TestRepository',
          Type: 'Tree',
          Branch: 'source',
          Url: 'https://github.com/owner/GitHubSync.TestRepository/tree/source'
        },
        RootTreePart: {
          Owner: 'owner',
          Repository: 'GitHubSync.TestRepository',
          Type: 'Tree',
          Branch: 'source',
          Url: 'https://github.com/owner/GitHubSync.TestRepository/tree/source'
        }
      }
    },
    {
      Parts: {
        Owner: 'owner',
        Repository: 'GitHubSync.TestRepository',
        Branch: 'source',
        Path: 'added2',
        Name: 'added2',
        NumberOfPathSegments: 1,
        Url: 'https://github.com/owner/GitHubSync.TestRepository/blob/source/added2',
        ParentTreePart: {
          Owner: 'owner',
          Repository: 'GitHubSync.TestRepository',
          Type: 'Tree',
          Branch: 'source',
          Url: 'https://github.com/owner/GitHubSync.TestRepository/tree/source'
        },
        RootTreePart: {
          Owner: 'owner',
          Repository: 'GitHubSync.TestRepository',
          Type: 'Tree',
          Branch: 'source',
          Url: 'https://github.com/owner/GitHubSync.TestRepository/tree/source'
        }
      },
      Target: 'target2'
    },
    {
      Parts: {
        Owner: 'owner',
        Repository: 'GitHubSync.TestRepository',
        Branch: 'source',
        Path: 'sourceDir/added3',
        Name: 'added3',
        NumberOfPathSegments: 2,
        Url: 'https://github.com/owner/GitHubSync.TestRepository/blob/source/sourceDir/added3',
        ParentTreePart: {
          Owner: 'owner',
          Repository: 'GitHubSync.TestRepository',
          Type: 'Tree',
          Branch: 'source',
          Path: 'sourceDir',
          Name: 'sourceDir',
          NumberOfPathSegments: 1,
          Url: 'https://github.com/owner/GitHubSync.TestRepository/tree/source/sourceDir',
          ParentTreePart: {
            Owner: 'owner',
            Repository: 'GitHubSync.TestRepository',
            Type: 'Tree',
            Branch: 'source',
            Url: 'https://github.com/owner/GitHubSync.TestRepository/tree/source'
          },
          RootTreePart: {
            Owner: 'owner',
            Repository: 'GitHubSync.TestRepository',
            Type: 'Tree',
            Branch: 'source',
            Url: 'https://github.com/owner/GitHubSync.TestRepository/tree/source'
          }
        },
        RootTreePart: {
          Owner: 'owner',
          Repository: 'GitHubSync.TestRepository',
          Type: 'Tree',
          Branch: 'source',
          Url: 'https://github.com/owner/GitHubSync.TestRepository/tree/source'
        }
      },
      Target: 'targetDir/target3'
    }
  ]
}